### PR TITLE
Add highlighted state/helper to combobox

### LINKED ISSUE #341: 🌟Feature request: add $isActive store
**Is your feature request related to a problem? Please describe.**
I'd love to build something like this:
image

Here, there is a clear distinction between `activeItem` and `selectedItem`. The first returns the item that's highlighted, so the values of the item can be used to render more information, while the `selectedItem` is used to set the actual `value` of the input.

**Describe the solution you'd like**
Add a store called `$isActive` that has the same consumer API as `$isSelected`

**Describe alternatives you've considered**
I still haven't thought potential alternatives all the way through because this feature is still a "nice to have" for my project for the time being.

**Additional context**
None

### PR DESCRIPTION
**Pull request by wesharper**
This adds the `isHighlighted` helper and a readable state called `highlighted`. The motivation is to enable more complex custom command palettes where the highlighted item changes the UI of a different surface like in this example:

image

- resolves #341

### CODE REVIEW

**Comment by wesharper** [comment]
Hmm... tests are all passing locally...

File: src/lib/builders/combobox/create.ts [hunk 1]
```diff
@@ -231,6 +235,16 @@ export function createCombobox<Value>(props?: CreateComboboxProps<Value>) {
 		return (item: Value) => deepEqual($value?.value, item);
 	});
 
+	/**
+	 * Determines if a given item is highlighted.
+	 * This is useful for displaying additional markup on the highlighted item.
+	 */
+	const isHighlighted = derived([highlightedItem], ([$value]) => {
+		return (item: Value) => {
+			return deepEqual($value ? getOptionProps($value).value : null, item);
```
Comment (TGlide) [inline src/lib/builders/combobox/create.ts:244]: Couldn't you just use `highlighted` here instead of `highlightedItem`?

**Comment by TGlide** [comment]
Hey @wesharper ! Thank you for the PR 🎉 

Just some small comments left :)

**Comment by TGlide** [comment]
> Hmm... tests are all passing locally...
> 
> I wonder if this is an async issue related to debounce? Have we seen intermittent failures on this in the past?

The Portal tests seem to fail now, for some reason. They work locally. I've yet to debug why

File: src/lib/builders/combobox/create.ts (hunk 1 above)
Comment (wesharper) [reply src/lib/builders/combobox/create.ts:244]: yep, I built isHighlighted first lol. Good call

**Comment by wesharper** [comment]
> 

Same, I'll have some time on Monday maybe. Sorry for the delay!

**Comment by wesharper** [comment]
Looks like all tests are passing reliably locally and on CI now @TGlide. Also, made the change you requested. Should be mergable

**Review by TGlide — APPROVED** [review]

**Pull request merged**